internal/compliance/snapshot_poller/pollers/aws/eks_cluster.go

```
@@ -0,0 +1,310 @@
+			NodeRole:       curNodegroup.NodeRole,
+			ReleaseVersion: curNodegroup.ReleaseVersion,
+			RemoteAccess:   curNodegroup.RemoteAccess,
+			Resources:      curNodegroup.Resources,
+			ScalingConfig:  curNodegroup.ScalingConfig,
+			Subnets:        curNodegroup.Subnets,
+			Version:        curNodegroup.Version,
+			// Normalized name for CreatedAt
+			TimeCreated: utils.DateTimeFormat(aws.TimeValue(curNodegroup.CreatedAt)),
+		})
+	}
+
+	return nodegroupResults, nil
+}
+
+// buildEksClusterSnapshot returns a complete snapshot of an EKS cluster
+func buildEksClusterSnapshot(eksSvc eksiface.EKSAPI, clusterName *string) (*awsmodels.EksCluster, error) {
+	if clusterName == nil {
+		return nil, nil
+	}
+
+	details, err := describeEKSCluster(eksSvc, clusterName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	eksCluster := &awsmodels.EksCluster{
+		GenericAWSResource: awsmodels.GenericAWSResource{
+			ARN:  details.Arn,
+			Name: details.Name,
+			Tags: details.Tags,
+		},
+		GenericResource: awsmodels.GenericResource{
+			ResourceID:   details.Arn,
+			ResourceType: aws.String(awsmodels.EksClusterSchema),
+			TimeCreated:  utils.DateTimeFormat(aws.TimeValue(details.CreatedAt)),
+		},
+		CertificateAuthority: details.CertificateAuthority,
+		EncryptionConfig:     details.EncryptionConfig,
+		Endpoint:             details.Endpoint,
+		Identity:             details.Identity,
+		Logging:              details.Logging,
+		PlatformVersion:      details.PlatformVersion,
+		ResourcesVpcConfig:   details.ResourcesVpcConfig,
+		RoleArn:              details.RoleArn,
+		Status:               details.Status,
+		Version:              details.Version,
+	}
+
+	eksCluster.FargateProfile, err = getEKSFargateProfiles(eksSvc, details.Name)
+	if err != nil {
+		zap.L().Warn("Error with IAM Permissions - Use the AWS Console or CloudFormation to update the Panther" +
+			"Audit Role policy to include the eks:ListFargateProfiles and eks:DescribeFargateProfile permissions")
+	}
```

Lets only catch the error & continue if it's specifically the auth error. If it's not an auth error, we want to return the error like normal.

This can be done by comparing it to a constant defined by the package (preferred) such as here: https://github.com/panther-labs/panther/blob/master/internal/compliance/snapshot_poller/pollers/aws/iam_user.go#L159-L160. Sadly this relies on the AWS service actually defining the error codes in the package, which is not always the case.

If the error codes are not defined in the package, this can be done by doing string comparisons on the error message such as here: https://github.com/panther-labs/panther/blob/master/internal/compliance/snapshot_poller/pollers/aws/cloudformation_stack.go#L133-L134.

Let me know if you have any questions on this.